src/tools/collect-license-metadata/src/reuse.rs

```diff
@@ -0,0 +1,49 @@
+use crate::licenses::{License, LicenseId, LicensesInterner};
+use anyhow::Error;
+use std::path::{Path, PathBuf};
+use std::process::{Command, Stdio};
+use std::time::Instant;
+
+pub(crate) fn collect(
+    reuse_exe: &Path,
+    interner: &mut LicensesInterner,
+) -> Result<Vec<(PathBuf, LicenseId)>, Error> {
+    eprintln!("gathering license information from REUSE");
+    let start = Instant::now();
+    let raw = &obtain_spdx_document(reuse_exe)?;
+    eprintln!("finished gathering the license information from REUSE in {:.2?}", start.elapsed());
+
+    let document = spdx_rs::parsers::spdx_from_tag_value(&raw)?;
+
+    let mut result = Vec::new();
+    for file in document.file_information {
+        let license = interner.intern(License {
```

Comment:
as someone unfamiliar with SPDX and with the overall plans here: What is the expected interning behavior when the file has differing `copyright_text` but the same `concluded_license` ? Are those supposed to be interned to the same LicenseId, or to distinct LicenseId's?

Reply:
Same license but different copyright holder would result in different `LicenseId`s, as that's effectively a different license (since most licenses require attribution, and the copyright statement is the attribution).